src/layout.rs

```diff
@@ -270,7 +270,7 @@ impl<T, K> WaitingState<T, K> {
             .iter()
             .find(|s| self.is_corresponding_release(&s.event))
         {
-            if self.timeout >= self.delay - since {
+            if since > self.delay || self.timeout >= self.delay - since {
```

Comment:
Does it works by changing the substraction by an addition?

```suggestion
            if self.timeout + since >= self.delay {
```

Reply:
It's a good suggestion. I'll try it more extensively in 10 days.

Reply:
I have not found any issue with the new version of the patch.